src/connector/src/parser/common.rs

```diff
@@ -32,3 +37,119 @@ pub(crate) fn json_object_smart_get_value<'a, 'b>(
     }
     None
 }
+
+pub fn mysql_row_to_datums(mysql_row: &mut MysqlRow, schema: &Schema) -> Vec<Datum> {
+    let mut datums = vec![];
+    for i in 0..schema.fields.len() {
+        let rw_field = &schema.fields[i];
+        let datum = {
+            match rw_field.data_type {
+                DataType::Boolean => {
+                    let v = mysql_row.take::<bool, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Int16 => {
+                    let v = mysql_row.take::<i16, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Int32 => {
+                    let v = mysql_row.take::<i32, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Int64 => {
+                    let v = mysql_row.take::<i64, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Float32 => {
+                    let v = mysql_row.take::<f32, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Float64 => {
+                    let v = mysql_row.take::<f64, _>(i);
+                    v.map(ScalarImpl::from)
+                }
+                DataType::Decimal => {
+                    let v = mysql_row.take::<RustDecimal, _>(i);
+                    v.map(|v| ScalarImpl::from(Decimal::from(v)))
+                }
+                DataType::Varchar => {
+                    // snapshot data doesn't contain offset, just fill None
+                    if rw_field.name.as_str() == OFFSET_COLUMN_NAME {
+                        None
+                    } else {
+                        let v = mysql_row.take::<String, _>(i);
+                        v.map(ScalarImpl::from)
+                    }
+                }
+                DataType::Date => {
+                    let v = mysql_row.take::<NaiveDate, _>(i);
+                    v.map(|v| ScalarImpl::from(Date::from(v)))
+                }
+                DataType::Time => {
+                    let v = mysql_row.take::<chrono::NaiveTime, _>(i);
+                    v.map(|v| ScalarImpl::from(Time::from(v)))
+                }
+                DataType::Timestamp => {
+                    let v = mysql_row.take::<chrono::NaiveDateTime, _>(i);
+                    v.map(|v| ScalarImpl::from(Timestamp::from(v)))
+                }
+                _ => unimplemented!("unsupported data type: {:?}", rw_field.data_type),
```

Comment:
- `unimplemented!` will panic the whole CN. Do you have any validation to make sure the code path won't be hit? Otherwise please use `Err` instead.
- We should support JSON as well. Please also test `TINYINT`. There is no reason that we only support part of the data types unless there is no corresponding data tpye in RisingWave. @cyliu0 is working on a testing framework to ensure all data types are correctly supported.

Reply:
I've add more data types, in theory all MySQL data types can be ingested. @xiangjinwu  Btw could you review the data type handling procedure in `src/connector/src/parser/mysql.rs`?

Reply:
Current data type mapping lgtm